src/Sylius/Component/Core/Shipping/Calculator/FlatRateCalculator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Shipping\Calculator;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Shipping\Calculator\CalculatorInterface;
+use Sylius\Component\Shipping\Model\ShipmentInterface;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+final class FlatRateCalculator implements CalculatorInterface
+{
+    /**
+     * @var CalculatorInterface
+     */
+    private $calculator;
+
+    /**
+     * @var ChannelContextInterface
+     */
+    private $channelContext;
+
+    /**
+     * @param CalculatorInterface $calculator
+     * @param ChannelContextInterface $channelContext
+     */
+    public function __construct(CalculatorInterface $calculator, ChannelContextInterface $channelContext)
+    {
+        $this->calculator = $calculator;
+        $this->channelContext = $channelContext;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function calculate(ShipmentInterface $subject, array $configuration)
+    {
+        $channel = $this->channelContext->getChannel();
```

Comment:
Ok, but `$subject` is a `ShipmentInterface` which doesn't have `getOrder` method..